Formula/d/dotnet.rb

```diff
@@ -39,62 +36,32 @@ class Dotnet < Formula
   # GCC builds have limited support via community.
   fails_with :gcc
 
-  # Backport fix for error on aspnetcore version while building 'installer in tarball'.
-  # TODO: Remove when available in release.
-  # PR ref: https://github.com/dotnet/installer/pull/14938
-  patch do
-    url "https://raw.githubusercontent.com/Homebrew/formula-patches/f206f7a45b330cce79e6bfe9116fccd93b0d3ed8/dotnet/aspnetcore-version.patch"
-    sha256 "00103452e2f52831c04007f1b7f9fcd5ecddf0671943657104f0ac8d3a9ca613"
-  end
-
-  # Fix build failure on macOS due to missing bootstrap packages
-  # Fix build failure on macOS ARM due to `osx-x64` override
-  # Issue ref: https://github.com/dotnet/source-build/issues/2795
-  patch :DATA
-
   def install
-    if OS.linux?
+    if OS.mac?
+      # Disable crossgen2 optimization in ASP.NET Core to work around build failure trying to find tool.
+      # Microsoft.AspNetCore.App.Runtime.csproj(445,5): error : Could not find crossgen2 tools/crossgen2
+      # TODO: Try to remove in future .NET 8 release or when macOS is officially supported in .NET 9
+      inreplace "src/aspnetcore/src/Framework/App.Runtime/src/Microsoft.AspNetCore.App.Runtime.csproj",
+                "<CrossgenOutput Condition=\" '$(TargetArchitecture)' == 's390x'",
+                "<CrossgenOutput Condition=\" '$(TargetOsName)' == 'osx'"
+    else
       ENV.append_path "LD_LIBRARY_PATH", Formula["icu4c"].opt_lib
-      ENV.append_to_cflags "-I#{Formula["krb5"].opt_include}"
-    end
-
-    # The source directory needs to be outside the installer directory
-    (buildpath/"installer").install buildpath.children
-    cd "installer" do
-      system "./build.sh", "/p:ArcadeBuildTarball=true", "/p:TarballDir=#{buildpath}/sources"
     end
 
-    cd "sources" do
-      # Use our libunwind rather than the bundled one.
-      inreplace "src/runtime/eng/SourceBuild.props",
-                "/p:BuildDebPackage=false",
-                "\\0 --cmakeargs -DCLR_CMAKE_USE_SYSTEM_LIBUNWIND=ON"
```

Comment:
May need to be verify if still need inreplace/patch. Some refactoring was done upstream so don't know which file is handling CMake args yet.